yfinance/data.py

```diff
@@ -39,22 +43,316 @@ def wrapped(*args, **kwargs):
 class TickerData:
     """
     Have one place to retrieve data from Yahoo API in order to ease caching and speed up operations
+
+    Credit for code for cookie & crumb goes to StackOverflow:
+    https://stackoverflow.com/questions/76065035/yahoo-finance-v7-api-now-requiring-cookies-python
     """
     user_agent_headers = {
         'User-Agent': 'Mozilla/5.0 (Macintosh; Intel Mac OS X 10_10_1) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/39.0.2171.95 Safari/537.36'}
 
     def __init__(self, ticker: str, session=None):
         self.ticker = ticker
-        self._session = session or requests
+        self._session = session or requests.Session()
 
-    def get(self, url, user_agent_headers=None, params=None, proxy=None, timeout=30):
-        proxy = self._get_proxy(proxy)
+        try:
+            self._session.cache
+        except AttributeError:
+            # Not caching
+            self._session_is_caching = False
+        else:
+            # Is caching. This is annoying. 
+            # Can't simply use a non-caching session to fetch cookie & crumb, 
+            # because then the caching-session won't have cookie.
+            self._session_is_caching = True
+            from requests_cache import DO_NOT_CACHE
+            self._expire_after = DO_NOT_CACHE
+        if self._session_is_caching and utils.cookie is None:
+            utils.print_once("WARNING: cookie & crumb does not work well with requests_cache. Am experimenting with 'expire_after=DO_NOT_CACHE', but you need to help stress-test.")
+
+        # Default to using 'basic' strategy
+        self._cookie_strategy = 'basic'
+        # If it fails, then fallback method is 'csrf'
+        # self._cookie_strategy = 'csrf'
+
+    def _toggle_cookie_strategy(self):
+        if self._cookie_strategy == 'csrf':
+            utils.get_yf_logger().debug(f'toggling cookie strategy {self._cookie_strategy} -> basic')
+            self._session.cookies.clear()
```

Comment:
Hey, I am getting this error on this branch:
```
    if ticker.info and "ask" in ticker.info and ticker.info["ask"] and is_number(ticker.info["ask"]):
  File "/usr/local/lib/python3.9/site-packages/yfinance/ticker.py", line 142, in info
    return self.get_info()
  File "/usr/local/lib/python3.9/site-packages/yfinance/base.py", line 1736, in get_info
    data = self._quote.info
  File "/usr/local/lib/python3.9/site-packages/yfinance/scrapers/quote.py", line 571, in info
    self._fetch(self.proxy)
  File "/usr/local/lib/python3.9/site-packages/yfinance/scrapers/quote.py", line 601, in _fetch
    result = self._data.get_raw_json(
  File "/usr/local/lib/python3.9/site-packages/yfinance/data.py", line 373, in get_raw_json
    response = self.get(url, user_agent_headers=user_agent_headers, params=params, proxy=proxy, timeout=timeout)
  File "/usr/local/lib/python3.9/site-packages/yfinance/utils.py", line 117, in wrapper
    result = func(*args, **kwargs)
  File "/usr/local/lib/python3.9/site-packages/yfinance/data.py", line 349, in get
    self._toggle_cookie_strategy()
  File "/usr/local/lib/python3.9/site-packages/yfinance/data.py", line 80, in _toggle_cookie_strategy
    self._session.cookies.clear()
AttributeError: module 'requests.cookies' has no attribute 'clear'
```

Reply:
Update